### PR TITLE
fix: filter site-disabled models in token-candidates

### PR DESCRIPTION
**Pull request by dorokuma**
## 问题

在站点设置中禁用了某个模型后，前端路由配置页面依然会把该模型提示为“待注册”或待补令牌。

排查发现 `/api/models/token-candidates` 在统计候选模型和缺失令牌时，只查了 `model_availability` 和 `token_model_availability`，没有读取 `site_disabled_models` 做过滤。而在实际构建路由时，`modelService.rebuildTokenRoutesFromAvailability` 已经按 `site_disabled_models` 排除了这些模型，导致统计接口和实际路由构建逻辑不一致。

## 改动

在 `/api/models/token-candidates` 中读取 `site_disabled_models`，对齐 `modelService` 的过滤逻辑，过滤以下 4 处消费点：

1. `models`（已覆盖模型列表）
2. `modelsWithoutToken`（缺失令牌候选）
3. `accountIdsForGroupHints`（分组提示账号集合）
4. `modelsMissingTokenGroups`（缺失分组令牌候选）

## 验证

- 新增 `src/server/routes/api/stats.token-candidates.test.ts` 测试用例，验证站点禁用模型后不会出现在候选与待补令牌列表中，同站点/同账号下其它未禁用模型正常保留
- `npm test -- src/server/routes/api/stats.token-candidates.test.ts` 测试通过
- `npm run typecheck:server` 与 `npm run typecheck:web` 通过
- `npm test` 全量测试通过

## Additional fixes (addressing CodeRabbit review)

- Deduplicated the site-disabled filtering into shared exported helpers in `modelService` (per AGENTS.md one-source-of-truth; both the token-candidates route and `rebuildTokenRoutesFromAvailability` now use it).
- Normalized disabled model names with `trim().toLowerCase()` on both sides so whitespace-padded rows (e.g. restored from backup) no longer silently bypass the filter. Edge-case tests added.

## Not addressed (intentional)

- `backupService` restore-side normalization and the case-sensitive display flag in `accounts.ts` are intentionally left out of this PR (display semantics vs filtering; read-side defense already covers the filter path).

## Summary by CodeRabbit

- **Bug Fixes**
  - Disabled models are excluded from token-candidate results across all response collections.
  - Disabled-model matching now consistently handles capitalization and surrounding whitespace.
  - Active models retain expected coverage, including sites without disabled models.
  - Improved reasoning and content parsing for streaming and non-streaming responses, including think-tag variants and direct reasoning events.
  - Prevented duplicated reasoning from appearing as assistant content.

- **Documentation**
  - Added development, contribution, and technical decision-recording guidance for project contributors.

### CODE REVIEW

File: src/server/routes/api/stats.ts [hunk 1]
```diff
@@ -1468,6 +1468,27 @@ export async function statsRoutes(app: FastifyInstance) {
...
+        disabledModelsBySite.get(row.siteId)!.add(row.modelName.toLowerCase());
+      }
+
+      function isModelDisabledForSite(
+        siteId: number,
+        modelName: string,
+      ): boolean {
+        const disabled = disabledModelsBySite.get(siteId);
+        return !!disabled && disabled.has(modelName.toLowerCase());
+      }
```
[the remark this answers is not in the document]
Comment (dorokuma) [reply src/server/routes/api/stats.ts]: Addressed in f220f5e: filtering logic now lives in shared exported helpers in modelService.ts; the route only loads and delegates.

File: src/server/routes/api/stats.ts [hunk 2]
```diff
@@ -1468,6 +1468,27 @@ export async function statsRoutes(app: FastifyInstance) {
...
+      const disabledModelRows = await db
+        .select()
+        .from(schema.siteDisabledModels)
+        .all();
+      const disabledModelsBySite = new Map<number, Set<string>>();
+      for (const row of disabledModelRows) {
+        if (!disabledModelsBySite.has(row.siteId)) {
+          disabledModelsBySite.set(row.siteId, new Set());
+        }
+        disabledModelsBySite.get(row.siteId)!.add(row.modelName.toLowerCase());
```
[the remark this answers is not in the document]
Comment (dorokuma) [reply src/server/routes/api/stats.ts]: Addressed in f220f5e: both index building and the predicate normalize with trim().toLowerCase(); edge-case tests added.
